file/file.go

```diff
@@ -404,23 +404,40 @@ func Glob(pth string, opt *Options) ([]stat.Stats, error) {
 	return glbSts, nil
 }
 
+// matchFolder resolves and returns directories that match a globbed path prefix.
+//
+// Behavior:
+//   - Splits the provided path into a parent directory and a final segment.
+//   - Recursively expands any glob/meta characters ([], *, ?) that appear in the
+//     parent directory portion, producing a concrete list of parent directories.
+//   - For each resolved parent, lists its entries and matches only subdirectories
+//     against the final segment. The final segment can be a literal or a pattern;
+//     literal is compared by equality, patterns use filepath.Match.
+//   - Returns the directories that match the final segment as stat.Stats entries.
+//
+// This function is used by Glob to support nested directory globs such as:
+//
+//	"./a/*/b[1-3]/*/file.txt"
+//
+// ensuring trailing directory segments are preserved and matched correctly.
 func matchFolder(pth string, opt *Options) (folders []stat.Stats, err error) {
-	pthDir, pattern := path.Split(strings.TrimRight(pth, "/"))
-	paths := []string{pthDir}
+	// Resolve patterns in the parent directory first, then match the final segment
+	parentDir, segment := path.Split(strings.TrimRight(pth, "/"))
```

Comment:
What is parent in this context? pthDir is in reference to the param `pth string`